src/services/sync/strategies/LDAPSyncerConsumer.js

```diff
@@ -99,74 +95,54 @@ class LDAPSyncerConsumer {
 			updateObject.roles = user.roles;
 		}
 		if (!_.isEmpty(updateObject)) {
-			this.app.service('users').patch(userData._id, updateObject);
-			accountModel.update(
-				{ userId: userData._id, systemId: account.systemId },
-				{
-					username: account.username,
-					userId: userData._id,
-					systemId: account.systemId,
-					activated: true,
-				},
-				{ upsert: true }
-			);
+			this.userRepo.patch(userData._id, updateObject);
+			this.accountRepo.update(userData._id, account);
 		}
 	}
 
-	createUser(idmUser, account, school) {
-		this.app
-			.service('users')
-			.create({
-				firstName: idmUser.firstName,
-				lastName: idmUser.lastName,
-				schoolId: school._id,
-				email: idmUser.email,
-				ldapDn: idmUser.ldapDn,
-				ldapId: idmUser.ldapId,
-				roles: idmUser.roles,
-			})
-			.then((user) =>
-				accountModel.create({
-					userId: user._id,
-					username: account.username.toLowerCase(),
-					systemId: account.systemId,
-					activated: true,
-				})
-			)
-			.catch((err) => logger.error('LDAP SYNC: error while creating User', err));
+	async createUser(idmUser, account, schoolId) {
+		try {
+			idmUser.schoolId = schoolId;
+			const user = await this.userRepo.create(idmUser);
+			await this.accountRepo.create(user._id, account);
+		} catch (err) {
+			logger.error('LDAP SYNC: error while creating User', err);
```

Comment:
We should think about it. Generally the error message should throw in the business layer and not in replo layer. For ldap operations we want a action list what is done step by step with error or not. In this case i think it make sense to bring the logging to the repo layer. 

The second is that we should not log and throw in the same step. 
We throw or we log and we should not log in deeper levels only on the final catch points.
But we must look deeper together how we can resort it.